src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -746,14 +883,90 @@ struct MHASingleToken {
                     const PlainTensor& attention_mask,
                     PlainTensor& output_emb,
                     const PlainTensor& beams,
+                    size_t max_context_len,
                     const PlainTensor& context_lens,
                     bool has_out_transpose,
                     bool auto_causal,
                     float d_scale,
                     const PlainTensor& k_scale_zp,
                     const PlainTensor& v_scale_zp) {
-        mha_single_token(query, present_key, present_value, alibi_mask, attention_mask, beams, context_lens, output_emb,
-            m_attn_w, m_temp, has_out_transpose, auto_causal, d_scale, k_scale_zp, v_scale_zp, m_head_sum);
+        auto B = query.size(0);
+        auto H = query.size(1);
+        auto q_len = query.size(2);
+        bool is_pagedattn = context_lens;
+        size_t kv_len;
+        if (is_pagedattn) {
+            kv_len = max_context_len;
+        } else {
+            kv_len = present_key.size(2);
+        }
+
+        bool fastpath_valid = false;
+#ifdef OPENVINO_ARCH_X86_64
+        if (is_pagedattn) {
+            auto S = query.size(3);
+            size_t block_size = present_value.size(2);
+            fastpath_valid = mayiuse(amx_bf16) && (S % 32 == 0) && (block_size % 16 == 0) && (S <= 32 * 6) && present_key.get_precision() == ov::element::bf16;
+            if (fastpath_valid) {
+                m_attn_w.resize<float>({B, H, q_len, (kv_len + block_size - 1) / block_size * block_size});
+                if (!m_gemv)
+                    m_gemv = std::make_shared<JitMatMulVecAMX>(static_cast<int>(S), static_cast<int>(block_size));
+                auto h_group_num = present_value.size(1);
+                size_t h_each_group_len = 1;
+                if (h_group_num != H) {
+                    h_each_group_len = H / h_group_num;
+                }
+                auto kv_len_in_blocks = beams.m_dims[1];
+                auto nthr = static_cast<size_t>(parallel_get_max_threads());
+                size_t real_len = 0;
+                for (size_t b = 0; b < B; b++)
+                    real_len += static_cast<size_t>(context_lens.ptr<int32_t>()[b]) / block_size;
+                if (real_len > nthr) {
```

Comment:
Just for the record, this branch is trying to mitigate `auto_partitioner`'s limitation in cache-affinity by grouping calculations on one head into single work item,  if total kv_len sum is big enough to be paralleled.